preview-maker.js

```diff
@@ -1,184 +1,392 @@
+      node.setAttribute(`fill`, cssColor);
     }
 
     if (node.childNodes) {
-        for (let child of Array.from(node.childNodes)) {
-            if (child.setAttribute) {
-                fill(child, color);
-            }
+      for (let child of Array.from(node.childNodes)) {
+        if (child.setAttribute) {
+          innerFill(child, cssColor);
         }
+      }
     }
+  };
+
+  innerFill(rootNode);
 };
 
 const makePrev = async (themeBuffer, themeName, themeAuthor, template) => {
-    let theme;
-
-    if (themeBuffer instanceof Buffer) {
-        theme = new Attheme(themeBuffer.toString(`binary`));
-    } else {
-        theme = themeBuffer;
+  let theme, accentHue;
+
+  if (themeBuffer instanceof Buffer) {
+    theme = new Attheme(themeBuffer.toString(`binary`));
+  } else {
+    theme = themeBuffer;
+  }
+
+  const preview = parser.parseFromString(templates[template]);
+  const AVATAR_VARIABLES = [
+    `avatar_backgroundRed`,
+    `avatar_backgroundOrange`,
+    `avatar_backgroundViolet`,
+    `avatar_backgroundGreen`,
+    `avatar_backgroundCyan`,
+    `avatar_backgroundBlue`,
+  ];
+  const windowBackgroundWhite =
+    theme[`windowBackgroundWhite`] ||
+    defaultVariablesValues[`windowBackgroundWhite`];
+
+  const inBubble =
+    theme[`chat_inBubble`] || defaultVariablesValues[`chat_inBubble`];
+  const outBubble =
+    theme[`chat_outBubble`] || defaultVariablesValues[`chat_outBubble`];
+  if (Color.brightness(inBubble) > Color.brightness(outBubble)) {
+    theme[`chat_{in/out}Bubble__darkest`] = inBubble;
+  } else {
+    theme[`chat_{in/out}Bubble__darkest`] = outBubble;
+  }
+  let colors = [];
+  for (const variable in defaultVariablesValues) {
+    if (variable === `chat_wallpaper` && !theme.chat_wallpaper) {
+      continue;
     }
-
-    const preview = parser.parseFromString(templates[template]);
-
-    const inBubble = (
-        theme[`chat_inBubble`]
-        || defaultVariablesValues[`chat_inBubble`]
+    const elements = getElementsByClassName(preview, variable);
+    const color = theme[variable] || defaultVariablesValues[variable];
+    for (const element of elements) {
+      fill(element, color);
+    }
+    const chooseHsl = rgbToHsl(color);
+    let hueDifference = getDifference(chooseHsl, windowBackgroundWhite, `hue`);
+    let saturationDifference = getDifference(
+      chooseHsl,
+      windowBackgroundWhite,
+      `saturation`
     );
-    const outBubble = (
-        theme[`chat_outBubble`]
-        || defaultVariablesValues[`chat_outBubble`]
+    let lightnessDifference = getDifference(
+      chooseHsl,
+      windowBackgroundWhite,
+      `lightness`
     );
-
-    if (Color.brightness(inBubble) > Color.brightness(outBubble)) {
-        theme['chat_{in/out}Bubble__darkest'] = inBubble;
+    if (
+      hueDifference > 2 &&
+      chooseHsl.saturation > 0.04 &&
+      saturationDifference > 0.02
+    ) {
+      colors.push(Math.round(chooseHsl.hue));
+    }
+  }
+
+  if (colors.length) {
```

Comment:
```suggestion
  if (colors.length > 0) {
```

When I see a check like `if (foo)`, I read it as `if foo is not null or undefined...`, not as `if foo is not zero...`. please don't abuse the fact that `0` is a falsy value.